### PR TITLE
Allow fallback to pulling cert when checking private/public key consistency

### LINKED ISSUE #430: Fallback to fetching public key from certificate not always applied?
(moving this from a comment on a closed issue to a new thread)
Request: Make the fallback to fetching public key from certificate work with OpenSSL's SSL_CTX_use_Privatekey().

This fallback was apparently implemented by #129 but that does not seem to work when an EC private key is passed to `SSL_CTX_use_Privatekey()` in OpenSSL even if matching certificate is present with same ID. In this case `p11prov_ec_match()` gets called which fails to find the public key via `find_associated_obj()`. Would be nice to get the fallback applied here too for wider usability. 

Whereas, when signing using "openssl pkeyutl" the fallback does work and appears to follow a different code path.

Tested using git master branch and a soft token (softhsm2) where only private key (EC) and certificate were imported into the token using `p11tool` -- public key was not explicitly imported. The same token works with `pkcs11-helper` library. 

Unlike `p11tool`, `softhsm2-util` imports keypair in one step and creates both public and private key objects in the token --- in that case there are no such errors, obviously.

### PR DESCRIPTION
**Pull request by simo5**
#### Description

This behavior is needed to pass some openssl consistency checks with tokens that store the public key only in a cert object.

Fixes #430 

#### Checklist

- [x] Code modified for feature
- [x] Test suite updated with functionality tests
- [x] Test suite updated with negative tests
- ~[ ] Documentation updated~

#### Reviewer's checklist:

- [x] Any issues marked for closing are addressed
- [x] There is a test suite reasonably covering new functionality or modifications
- [x] This feature/change has adequate documentation added
- [x] Code conform to coding style that today cannot yet be enforced via the check style test
- [x] Commits have short titles and sensible commit messages
- [x] Coverity Scan has run if needed (code PR) and no new defects were found

### CODE REVIEW

File: src/objects.c [hunk 1]
```diff
@@ -2313,7 +2388,10 @@ static int match_public_keys(P11PROV_OBJ *key1, P11PROV_OBJ *key2)
     if (!assoc_pub_key) {
         P11PROV_raise(priv_key->ctx, CKR_GENERAL_ERROR,
                       "Could not find associated public key object");
-        return RET_OSSL_ERR;
+
+        /* some tokens only store the public key o a cert and not in a separate
```
Comment (Jakuje) [inline src/objects.c:2392]:
```suggestion
        /* some tokens only store the public key in a cert and not in a separate
```
?

File: tests/tlssetkey.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/* Copyright (C) 2022 Simo Sorce <simo@redhat.com>
```
Comment (Jakuje) [inline tests/tlssetkey.c:1]:
```suggestion
/* Copyright (C) 2024 Simo Sorce <simo@redhat.com>
```

File: tests/ttls [hunk 3]
```diff
@@ -7,6 +7,12 @@ source "${TESTSSRCDIR}/helpers.sh"
 title PARA "Test SSL_CTX creation"
 $CHECKER "${TESTBLDDIR}/tlsctx"
 
+if [ -n "$ECBASE2URI" ]; then
+    title PARA "Test setting cert/keys on TLS Context"
+    $CHECKER "${TESTBLDDIR}/tlssetkey" "${ECCRT2URI}" "${ECPRI2URI}"
+fi
```
Comment (Jakuje) [inline tests/ttls:13]: would it make sense to add a sanity test with full keys? Either in else branch or just with normal keys? And with RSA keys?

Comment (simo5) [reply tests/ttls:13]: RSA keys always have the public exponent so we'd be testing nothing really, not sure we need any other test really, this specific behavior with EC keys is the corner case that needs testing, other fallback behaviors are already tested independently as needed, and the general behavior is also already working.

**Comment by simo5** [comment]
Ok added test with URIs for private key that has the corresponding pub key too, it cost us nothing after all.

**Comment by simo5** [comment]
@Jakuje given you told me (privately) that there were no makor issues with this PR, all test pass and cosvcan is happy, I am going to merge.

I need this in for my next PR.
